src/providers/http.ts

```diff
@@ -1171,20 +1170,34 @@ export function determineRequestBody(
   configBody: Record<string, any> | any[] | string | undefined,
   vars: Record<string, any>,
 ): Record<string, any> | any[] | string {
+  // Parse stringified JSON body if needed (handles legacy data saved as strings)
+  let actualConfigBody = configBody;
+  if (typeof configBody === 'string' && contentType) {
+    try {
+      actualConfigBody = JSON.parse(configBody);
+      logger.debug('[HTTP Provider] Parsed stringified config body to object');
+    } catch (err) {
+      // If parsing fails, it's probably a template string or non-JSON content, leave as-is
+      logger.debug(
+        `[HTTP Provider] Config body is a string that couldn't be parsed as JSON, treating as template: ${String(err)}`,
+      );
+    }
+  }
```

Comment:
can we just not send strings from the ui? is there any evidence that this is happening for legacy configs